### PR TITLE
Remove checks for null before freeing

### PR DESCRIPTION
**Pull request by qarkai**
Simplify related code.

### CODE REVIEW

File: src/image.cc [hunk 1]
```diff
@@ -429,25 +427,24 @@ void image_update_title(ImageWindow *imd)
...
 		{
-		lw_ident = g_strconcat(" (", lw->options.id, ")", NULL);
+		LayoutWindow *lw = layout_find_by_image(imd);
+		if (lw)
+			{
+			lw_ident = g_strconcat(" (", lw->options.id, ")", NULL);
+			}
 		}
 
 	title = g_strdup_printf("%s%s%s%s%s%s%s",
```
Comment (xsdg) [inline src/image.cc:440]: Can this just use `g_strconcat` also?

Comment (qarkai) [reply src/image.cc:440]: I'd use `GString` for `title`. Something like

```c++
if (!imd->top_window) return;

g_autoptr(GString) title = g_string_new(imd->title);

if (imd->image_fd) title = g_string_append(title, imd->image_fd->name);

if (imd->title_show_zoom)
	{
	g_autofree gchar *buf = image_zoom_get_as_text(imd);
	g_string_append_printf(title, " [%s]", buf);
	}

if (imd->collection && collection_to_number(imd->collection) >= 0)
	{
	const gchar *name = imd->collection->name;
	if (!name) name = _("Untitled");
	g_string_append_printf(title, _(" (Collection %s)"), name);
	}

if (imd->image_fd) title = g_string_append(title, " - ");
if (imd->title_right) title = g_string_append(title, imd->title_right);

if (options->show_window_ids)
	{
	LayoutWindow *lw = layout_find_by_image(imd);
	if (lw)
		{
		g_string_append_printf(title, " (%s)", lw->options.id);
		}
	}

gtk_window_set_title(GTK_WINDOW(imd->top_window), title->str);
```

What do you think?

Comment (xsdg) [reply src/image.cc:440]: That looks like a really good simplification!  Probably best for a separate PR, though

Comment (qarkai) [reply src/image.cc:440]: > Probably best for a separate PR, though

Sure

**Comment by caclark** [comment]
Implemented in commit 042f44a079f43ad8d0df414ab54d2902cfacfc50

**Pull request closed without merging**